ruby-challenges.rb

```diff
@@ -10,22 +10,86 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def filter_out (arr,str) 
+
+    arr.find_all { |i| i.include? str}
+
+end
+
+p filter_out(beverages_array, letter_o) #=> ["coffee", "soda water"]
+p filter_out(beverages_array, letter_t) #=> ["tea", "water", "soda water"]
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def hash_it_out hash
+
+    hash.values.flatten.sort { |a, b| a <=> b }
+
+end
+
```

Comment:
Awesome! Just one thing, you actually don't need that block after `.sort` at all!